gpytorch/distributions/multivariate_normal.py

```diff
@@ -154,8 +164,13 @@ def rsample(self, sample_shape=torch.Size(), base_samples=None):
             res = res.view(sample_shape + self.loc.shape)
 
         else:
+            covar_root = covar.root_decomposition().root
+
             # Make sure that the base samples agree with the distribution
-            if self.loc.shape != base_samples.shape[-self.loc.dim() :]:
+            if (
+                self.loc.shape != base_samples.shape[-self.loc.dim() :]
+                and covar_root.shape[-1] < base_samples.shape[-1]
```

Comment:
To make sure I fully understand: The higher level concern is that the covar root is not full rank, right? Is the idea that we initially get base samples for the full rank and leave those fixed during SAA optimization so if we use low-rank root decomps we need need to tolerate the fact that the root loses a rank between evaluations at different test points?

Reply:
I don't think that's a new concern; it was already there. This check is just ensuring that we still check for the base samples having the correct shape (e.g. we don't lose too many ranks during the SAA approximation), but that it's okay for the base samples to not match in the case where the covar root is large.

Reply:
Hmm so my understanding would be that we'd start off with base samples compatible with a full rank covar root, and then just use however many we need given the rank of the covar root. This seems to be the opposite of what you say - did you mean to say "it's okay for the base samples to not match in the case where the covar root is *large*." (in terms of rank)?

Reply:
Yes, this allows root decompositions of size `n x m` where `m > n` to be used.